compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -746,45 +737,11 @@ abstract class PrepJSInterop extends plugins.PluginComponent
             val path = annot.stringArg(1).fold[List[String]](Nil)(parsePath)
             JSNativeLoadSpec.Import(module, path)
 
-          case Some(annot) if annot.symbol == JSNameAnnotation =>
-            if (!scalaJSOpts.suppressMissingJSGlobalDeprecations) {
-              reporter.warning(annot.pos,
-                  "@JSName on top-level native JS classes and objects " +
-                  "(or native JS classes and objects inside Scala objects) " +
-                  "is deprecated, and should be replaced by @JSGlobal (with " +
-                  "the same meaning). This will be enforced in 1.0." +
-                  SuppressMissingJSGlobalDeprecationsMsg)
-            }
-
-            globalFromName
-
           case None =>
-            if (needsExplicitJSName) {
-              if (sym.isModuleClass) {
-                reporter.error(pos,
-                    "Native JS objects inside non-native objects must " +
-                    "have an @JSGlobal or @JSImport annotation")
-              } else {
-                // This should be an error, but we erroneously allowed that before
-                reporter.warning(pos,
-                    "Native JS classes inside non-native objects should " +
-                    "have an @JSGlobal or @JSImport annotation. " +
-                    "This will be enforced in 1.0.")
-              }
-            } else {
-              if (!scalaJSOpts.suppressMissingJSGlobalDeprecations &&
-                  !sym.isPackageObjectClass) {
-                reporter.warning(pos,
-                    "Top-level native JS classes and objects should have an " +
-                    "@JSGlobal or @JSImport annotation. This will be " +
-                    "enforced in 1.0.\n" +
-                    "  If migrating from 0.6.14 or earlier, the equivalent " +
-                    "behavior is an @JSGlobal without parameter." +
-                    SuppressMissingJSGlobalDeprecationsMsg)
-              }
-            }
+            reporter.error(pos, "Native JS classes and objects must have an " +
+                "@JSGlobal or @JSImport annotation")
```

Comment:
Also consider moving this check inside `checkAndGetJSNativeLoadingSpecAnnotOf`, and unify the error message with the case with duplicate loading spec annots.